app/services/secure_application_finder.rb

```diff
@@ -24,6 +26,10 @@ def check_for_errors
   end
 
   def expired?
-    secure_data[:expired_at] && secure_data[:expired_at] < Time.current
+    expires_on && expires_on < Time.current
+  end
+
+  def expires_on
+    legal_aid_application.citizen_url_expires_on.presence || secure_data[:expired_at]
```

Comment:
can we/should we be using  presence for the secure_data too

```suggestion
    legal_aid_application.citizen_url_expires_on.presence || secure_data[:expired_at].presence
```

Reply:
`object.presence` is another way of writing

```ruby
object.present? ? object : nil
```

so it wouldn't mean anything if we added it to `secure_data[:expired_at]`

https://api.rubyonrails.org/classes/Object.html#method-i-presence